interactive_sentiment_analysis.py

```diff
@@ -2,28 +2,44 @@
 import numpy as np
 import mmh3
 
-def demo(network: bolt.Network):
+def demo(network: bolt.Network, verbose: bool):
     include_bigrams = True
     k = 0
     seed = 0
     feat_hash_dim = 100000
+    
     while True:
         inp = input('Input:')
         line = inp.lower().strip()
+        if line == "exit":
+            print("Exiting demo...")
+            break
         itms = line.split(' ')
+        
         ## we can add stemming, lower casing etc here
         x_idxs = [mmh3.hash(itm, seed) % feat_hash_dim for itm in itms]
         if include_bigrams:
             x_idxs += [mmh3.hash(itms[j]+'-'+itms[j+1], seed)%feat_hash_dim for j in range(len(itms)-2)]
         if k>0:
             x_idxs += [mmh3.hash(line[j:j+k], seed)%feat_hash_dim for j in range(len(line)+1-k)]
-        ##
+        
+        ## Format as bolt numpy input
         x_offsets = np.int32([0,len(x_idxs)])
-        label = np.array([np.uint32(0)])
         x_vals = np.ones(len(x_idxs))
-        ######
-        temp = network.predict(x_idxs, x_vals, x_offsets, label, [0,1], batch_size=1)
-        if temp>0:
-            print('positive!')
+        y_idxs = np.array([np.uint32(0)])
+        y_vals = np.array([np.float32(1)])
+        y_offsets = np.array([np.uint32(10), np.uint32(10)])
+        
+        ## Predict
+        temp = network.predict(
```

Comment:
I think we would want to compare our prediction inference time with the pipeline() in line 248 in bolt.ipynb. So we would call `network.enable_sparse_inference` in this function before `predict()`. 
Also we should print the time used in inference when specified, which I assume is already your plan as you added a verbose flag in this function. 

Otherwise it looks good!